internal/app/cache/cache.go

```diff
@@ -29,8 +31,14 @@ type cache struct {
 	ttl     time.Duration
 }
 
+// Response stores the raw Discovery Response alongside its marshaled resources.
+type Response struct {
+	raw                v2.DiscoveryResponse
+	marshaledResources []*any.Any
```

Comment:
What is the reasoning behind storing both marshaled and raw? Why not just store just the raw response?

Reply:
To clarify, do you mean why not just store the marshaled response? (raw being the unserialized response and marshaled being the serialized resources)

Good question though! I don't have a strong opinion on either approach but my main thinking was retaining the raw response allows us to not have opaque logs in other areas of the call stack that may require logging the resources. What you said for debugging purposes! Differentiating between an upstream problem vs a serialization problem, etc.

Reply:
That makes sense to me. :)